src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -19,92 +19,135 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Configuration;
 using System.Diagnostics;
-using System.Web;
 using System.IO;
 using System.Text;
+using System.Web;
 
 public class PullRequestHandler : IHttpHandler
 {
-    private const string _gitExecutablePath = @"C:\Program Files\Git\cmd\git.exe";
-    private static string _gitRepoPath;
-    private const int _gitExecutionTimeoutMilliseconds = 100000;
-
-    public void ProcessRequest(HttpContext context)
-    {
-        // Check for prerequisites needed before this script can execute
+    private string _gitExecutablePath;
+    private string _gitRepositoriesBasePath;
+    private TimeSpan _gitExecutionTimeout;
 
-        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+    private bool _configurationRead = false;
 
-        StringBuilder result = new StringBuilder();
-        bool success = CheckPrerequisites(result);
+    private const string _gitExecutablePathConfigurationKeyString = "GitExecutablePath";
+    private const string _gitRepositoriesBasePathConfigurationKeyString = "GitRepositoriesBasePath";
+    private const string _gitExecutionTimeoutConfigurationKeyString = "GitExecutionTimeout";
 
-        if (!success)
-        {
-            // CheckPrerequisites failed, this is a server issue
-            context.Response.StatusCode = 500;
-            context.Response.Write(HttpUtility.HtmlEncode(result));
-            return;
-        }
+    private const string _gitRepoPathTemplate = @"{0}\wcf{1}";
 
-        success = CleanupMergedBranches(result);
+    public void ProcessRequest(HttpContext context)
+    {
+        StringBuilder result = new StringBuilder();
 
+        // Read configuration
+        bool success = ReadConfiguration(result);
         if (!success)
         {
-            // CleanupMergedBranches failed, this is a server issue
+            // ReadConfiguration failed, this is a server issue 
             context.Response.StatusCode = 500;
```

Comment:
I just throw ideas here, you decide if you want to take it or not...

After playing with PRService in the past a few days, many times I found it was hard to tell a 500 or a 400 are from IIS, ASP.NET or us. I wish we could throw a unique 500/400 number so I could tell more easily who failed the request.